daffodil-test/src/test/scala/org/apache/daffodil/runtime1/layers/CheckDigitLayer.scala

```diff
@@ -47,33 +47,37 @@ object CheckDigitLayer {
  *   - digitEncoding is a string which is the name of a character set encoding which
  *   defines the character set of the digits.
  *   This must match the dfdl:encoding for the DFDL element declaration using this layer.
- *
- * @param length the value of the length DFDL variable
- * @param params the value of the params DFDL variable
- * @param digitEncoding the value of the digitEncoding DFDL variable
  */
-class CheckDigitLayer(length: JInt, params: String, digitEncoding: String)
+class CheckDigitLayer
   extends ChecksumLayer(
     "checkDigit", // name of the layer, also happens to be the name of the output variable.
     CheckDigitLayer.ns,
-    length,
   ) {
 
+  private var params: String = _
+  private var digitEncoding: String = _
+  private lazy val charset = Charset.forName(digitEncoding)
+  private lazy val isVerbose = params.toLowerCase.contains("verbose")
+
+  /**
+   * @param length the value of the length DFDL variable
+   * @param params the value of the params DFDL variable
+   * @param digitEncoding the value of the digitEncoding DFDL variable
+   */
+  def setLayerVariableParameters(length: Int, params: String, digitEncoding: String): Unit = {
+    setLength(length)
+    this.params = params
+    this.digitEncoding = digitEncoding
+  }
+
   /**
    * Result DFDL variable value getter. This getter is called to obtain the value for,
    * and populate the DFDL variable named checkDigit, in the layer's target namespace.
    * @return the check digit value
    */
-  def getDFDLResultVariable_checkDigit: JInt =
+  def getLayerVariableResult_checkDigit: JInt =
```

Comment:
Just curious if this must be a Jint or would a primitive Int work too? I think it would work with the new changes? Should our examples use primitive int since I imagine that what most Java devs would find more natural?

Reply:
Question has not been resolved yet.

I got this far only by skimming; I didn't have time to review 2/3s of the PR thoroughly.  I will accept that the changes work since the tests pass.